### PR TITLE
Bug 1971162: OCS check if the disks are Installation eligible

### PR DESCRIPTION
**Pull request by rewantsoni**
Modify getValidDiskCount to check if the disks used are eligible for Installation before checking the diskSize and Type.

Closes: [OCPBUGSM-30837](https://issues.redhat.com/browse/OCPBUGSM-30837)
Signed-off-by: Rewant Soni <resoni@redhat.com>

# Description

The Virtal_Floppy had a disk type of HDD and it is not eligible for installation. So we used the InstallationEligible field in the Disk to check if the disk installed is eligible.

# What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [ ] Cloud
- [ ] Operator Managed Deployments
- [x] None

# How was this code tested?

Please, select one or more if needed:

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [ ] Waiting for CI to do a full test run
- [x] Manual (Elaborate on how it was tested)
- [ ] No tests needed

# Assignees

Please, add one or two reviewers that could help review this PR.

/assign @priyanka19-98 
/assign @avishayt 
/assign @nbalacha 
/assign @gobindadas

## Checklist

- [x] Title and description added to both, commit and PR.
- [x] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [x] Reviewers have been listed
- [x] This change does not require a documentation update (docstring, `docs`, README, etc)
- [x] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- [ ] Are the title and description (in both PR and commit) meaningful and clear?
- [ ] Is there a bug required (and linked) for this change?
- [ ] Should this PR be backported?

### CODE REVIEW

File: internal/operators/ocs/validations.go [hunk 1]
```diff
@@ -175,7 +176,11 @@ func (o *operator) getValidDiskCount(disks []*models.Disk, installationDiskID st
 	var countDisks int64
 	var err error
 
-	for _, disk := range disks {
+	eligibleDisks := funk.Filter(disks, func(disk *models.Disk) bool {
+		return disk.InstallationEligibility.Eligible
+	}).([]*models.Disk)
```
Comment (avishayt) [inline internal/operators/ocs/validations.go:181]: Why not just add this to the "if" below, to avoid traversing the list of disks an additional time?

Comment (rewantsoni) [reply internal/operators/ocs/validations.go:181]: Done

**Review by avishayt — APPROVED** [review]

**Pull request merged**